core/src/main/java/org/elasticsearch/rest/action/support/RestTable.java

```diff
@@ -79,7 +79,10 @@ public static RestResponse buildTextPlainResponse(Table table, RestChannel chann
             for (int col = 0; col < headers.size(); col++) {
                 DisplayHeader header = headers.get(col);
                 pad(new Table.Cell(header.display, table.findHeaderByName(header.name)), width[col], request, out);
-                out.append(" ");
+                boolean isLastColumn = col == headers.size()-1;
```

Comment:
Probably don't need to compute `headers.size() - 1` for every header, could move it out of the loop